lib/recent.go

```diff
@@ -0,0 +1,119 @@
+package lib
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type RecentFiles struct {
+	Terraform []string `json:"terraform"`
+	OpenTofu  []string `json:"openTofu"`
+}
+
+func addRecent(requestedVersion string, installPath string, distribution string) {
+	if !validVersionFormat(requestedVersion) {
+		logger.Errorf("The version %q is not a valid version string and won't be stored", requestedVersion)
+		return
+	}
+	installLocation := GetInstallLocation(installPath)
+	recentFilePath := filepath.Join(installLocation, recentFile)
+	var recentFileData RecentFiles
+	if CheckFileExist(recentFilePath) {
+		unmarshalRecentFileData(recentFilePath, &recentFileData)
+	}
+	prependRecentVersionToList(requestedVersion, installPath, distribution, &recentFileData)
+	saveRecentFile(recentFileData, recentFilePath)
+}
+
+func prependRecentVersionToList(version, installPath, distribution string, r *RecentFiles) {
+	var sliceToCheck []string
+	if distribution == distributionTerraform {
+		sliceToCheck = r.Terraform
+	} else if distribution == distributionOpenTofu {
+		sliceToCheck = r.OpenTofu
+	}
+	for versionIndex, versionValue := range sliceToCheck {
+		if versionValue == version {
+			sliceToCheck = append(sliceToCheck[:versionIndex], sliceToCheck[versionIndex+1:]...)
+		}
+	}
+	sliceToCheck = append([]string{version}, sliceToCheck...)
+
+	if len(sliceToCheck) > 3 {
+		deleteDownloadedBinaries(installPath, distribution, sliceToCheck[3:])
```

Comment:
Do we *want* to delete these binaries. If tfswitch sets an arbitrary limit of 3 versions, does that mean that user's should *only* cache 3 versions.
I'd had thought the recentVersions was _only_ a UI thing and not related to how many versions were being cached.

Reply:
For reference, at least for me, I use quite a few versions of terraform regularly (more than 3 for sure) and it's nice to always use local caches

Reply:
I agree. I wasn't aware that this was only a UI thing (why?) I thought this method missing. I can remove it

Reply:
I assume just to avoid cluttering up the UI. @warrensbox ?

Reply:
The RECENT version is for user to switch easily between their most used version.

RECENT is only a UI thing. We should not delete other version. It's nice to have the cache there.
@MatrixCrawler